www/js/metrics-factory.js

```diff
@@ -5,37 +5,61 @@ angular.module('emission.main.metrics.factory', ['emission.plugin.kvstore'])
 .factory('FootprintHelper', function() {
   var fh = {};
   var footprint = {
-    train: 92/1609,
-    car: 287/1609,
-    ON_FOOT: 0,
-    BICYCLING: 0
+    ON_FOOT:      0,
+    BICYCLING:    0,
+    CAR:        267/1609,
+    BUS:        278/1609,
+    TRAIN:       92/1609,
+    AIR_OR_HSR: 217/1609
   }
   var readable = function(v) {
     return v > 9999? Math.round(v / 1000) + 'k kg CO₂' : Math.round(v) + ' kg CO₂';
   }
   var mtokm = function(v) {
     return v / 1000;
   }
-  fh.getFootprintRaw = function(distance, mode) {
-    if (mode === "IN_VEHICLE") {
-      return [footprint.train * mtokm(distance), footprint.car * mtokm(distance)];
-    } else {
-      return footprint[mode] * mtokm(distance);
+
+  fh.readableFormat = function(v) {
+    return v > 9999? Math.round(v / 1000) + 'k kg CO₂' : Math.round(v) + ' kg CO₂';
+  }
+  fh.getFootprintForMetrics = function(userMetrics) {
+    var result = 0;
+    for (var i in userMetrics) {
+      var mode = userMetrics[i].key;
+      if (mode in footprint) {
+        result += footprint[mode] * mtokm(userMetrics[i].values);
+      }
+      else {
+        console.debug('WARNING: FootprintHelper.getFootprintFromMetrics() was requested for an unknown mode: ' + mode + " metrics JSON: " + JSON.stringify(userMetrics));
+      }
+    }
+    return result;
+  }
+  fh.getLowestFootprintForDistance = function(distance) {
+    var lowestFootprint = 9999;
+    for (var mode in footprint) {
+      if (mode == 'ON_FOOT' || mode == 'BICYCLING') {
+        // these modes aren't considered when determining the lowest carbon footprint
+      }
+      else {
+        lowestFootprint = Math.min(lowestFootprint, footprint[mode]);
```

Comment:
The main difference here is that instead of hardcoding that the lowest footprint is the train, you iterate over the modes and determine the lowest footprint from the data.

Reply:
Yes. So that if the data or dataset changes, the calculation still holds. Another option would be to pre-calculate and store this value only once (when it doesn't exist, e.g. the first time this screen is shown).

Reply:
I think that maybe you can document that as a potential future optimization, but we can move on now. premature optimization is the root of all evil.

Reply:
Agreed :)